### PR TITLE
Return error code when issue exists

### PR DESCRIPTION
**Pull request by ytet5uy4**
## WHY
When used tflint on CI, it's didn't failed although tflint return some errors.

## WHAT
Update to return 2 when issue exists.

### CODE REVIEW

**Comment by wata727** [comment]
Thanks for your proposal.
However, I think that it is not desirable to return the same status code as it was when an error occurred in core. Perhaps, it should return status code for that reason.

In my opinion, status code returned by lint tools should be customizable. We should be able to control it by runtime arguments as follows without changing the default behavior.

```
$ tflint
# return exit status code is 0 

$ tflint --exit-error-status
# return exit status code is something that is neither 0 nor 1
```

**Comment by ytet5uy4** [comment]
I've updated to match your opinion.
Is my understanding correct?

File: cli.go [hunk 1]
```diff
@@ -148,6 +150,12 @@ func (cli *CLI) Run(args []string) int {
 
 		p := printer.NewPrinter(cli.outStream, cli.errStream)
 		p.Print(issues, format)
+
+		if exitErrorStatus {
+			if len(issues) > 0 {
```
Comment (wata727) [inline cli.go]: [nitpicks] `if exitErrorStatus && len(issues) > 0`

File: cli.go [hunk 2]
```diff
@@ -71,6 +72,7 @@ func (cli *CLI) Run(args []string) int {
 	flags.StringVar(&awsAccessKey, "aws-access-key", "", "AWS access key used in deep check mode.")
 	flags.StringVar(&awsSecretKey, "aws-secret-key", "", "AWS secret key used in deep check mode.")
 	flags.StringVar(&awsRegion, "aws-region", "", "AWS region used in deep check mode.")
+	flags.BoolVar(&exitErrorStatus, "exit-error-status", false, "return exit status code when issue exists.")
```
Comment (wata727) [inline cli.go]: I'm sorry... Please use another option name 🙇 
I think `--exit-error-status` is bad name. (My recommendation is `--error-with-issues`)

File: cli.go [hunk 3]
```diff
@@ -148,6 +150,12 @@ func (cli *CLI) Run(args []string) int {
 
 		p := printer.NewPrinter(cli.outStream, cli.errStream)
 		p.Print(issues, format)
+
+		if exitErrorStatus {
+			if len(issues) > 0 {
+				return ExitCodeError
```
Comment (wata727) [inline cli.go]: Please use another status code. (ex: `ExitCodeIssuesFound`, etc.) `ExitCodeError` is reserved for execution error.

**Comment by wata727** [comment]
Great! Finally, Please squash to one commit as well :)

**Pull request merged**